chapters/chap03-sec03.tex

```diff
@@ -36,25 +36,25 @@ \section{\zeth~statement after primitive instantiation}\label{instantiation:stat
 
 \subsection{Instantiating the packing functions}\label{instantiation:statement:pack}
 
-As we consider SNARKs based on arithmetic circuits defined over a prime field, all variables in the constraint system are interpreted as field elements. Nevertheless, as illustrated~\cref{zeth-protocol:statement}, part of the statement consists in computing functions which co-domains are sets of binary strings (which may be longer than the bit representation of elements of the finite field). While a bit (i.e.~$\{0, 1\}$) is an element of $\FF_p$ ($p$ prime), it is important to minimize the number of gates in the arithmetic circuit for better performance for the proof generation and minimize the number of input wires to improve the verification time. This can be done by ``packing'' binary strings into a set of field elements by interpreting the binary sequence as the elements' base 2 decomposition. On the one hand, converting binary strings into field elements requires to extend the statement proven by the prover, and thus requires the addition of a few arithmetic gates. On the other hand, doing so allows to minimize the number of primary inputs, which, in fine, reduces the complexity of the SNARK verification carried out on-chain.
-Indeed, the cost of $\groth$ zk-SNARK~\cite{groth2016size} proof verification is linear in the number of primary inputs since each input acts as a scalar in a costly scalar multiplication of an element in $\gset_1$. Hence, while packing puts ``(slightly) more pressure'' on the prover -- by adding constraints in the circuit -- it simplifies the verifier's work.
+As we consider SNARKs based on arithmetic circuits defined over a prime field, all variables in the constraint system are interpreted as field elements. Nevertheless, as illustrated in~\cref{zeth-protocol:statement}, part of the statement consists of functions whose co-domains are sets of binary strings (which may be longer than the bit representation of elements of the finite field). While a bit (i.e.~$\{0, 1\}$) is an element of $\FF_p$ ($p$ prime), it is important to minimize the number of gates in the arithmetic circuit (for proof generation efficiency), and to minimize the number of input wires (to improve verification time). This can be done by representing fragments of binary strings as the base 2 decomposition of binary elements, thereby ``packing'' binary strings into multiple elements. Converting binary strings into field elements requires the addition of some arithmetic gates (extending the statement to be proven), but reduces the number of primary inputs (reducing the complexity of the SNARK verification carried out on-chain).
```

Comment:
`This can be done by representing fragments of binary strings as the base 2 decomposition of binary elements` -> `This can be done by representing fragments of binary strings as the base 2 decomposition of field elements`?

Reply:
Changed as suggested.